src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -131,156 +257,1263 @@ static DamageTable D_8089E314 = {
+    } else if (!func_801690CC(globalCtx)) {
+        temp_v0_2 = this->unk_28E + this->actor.shape.rot.y;
+        temp_v0_2 = BINANG_SUB(this->actor.yawTowardsPlayer, temp_v0_2);
+        temp_v0_2 = CLAMP(temp_v0_2, -0x300, 0x300);
+        this->unk_28E += temp_v0_2;
+        this->unk_28E = CLAMP(this->unk_28E, -0x2C00, 0x2C00);
+    }
+}
+
+s32 func_8089D60C(EnDinofos* this, GlobalContext* globalCtx) {
+    s32 i;
+
+    if (this->colliderJntSph.base.acFlags & AC_HIT) {
+        this->colliderJntSph.base.acFlags &= ~AC_HIT;
+        func_800BE2B8(&this->actor, &this->colliderJntSph);
+
+        for (i = 0; i < ARRAY_COUNT(this->colliderJntSphElement); i++) {
+            if (this->colliderJntSph.elements[i].info.bumperFlags & BUMP_HIT) {
+                break;
+            }
+        }
+
+        if (i == ARRAY_COUNT(this->colliderJntSphElement)) {
+            return false;
+        }
+
+        if ((this->unk_28A == 10) && (this->colliderJntSph.elements[i].info.acHitInfo->toucher.dmgFlags & 0xDB0B3)) {
+            return false;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 0xF) {
+            return false;
+        }
+
+        if (!func_800BE22C(&this->actor)) {
+            func_800BBA88(globalCtx, &this->actor);
+            D_8089E350--;
+            if (D_8089E350 == 0) {
+                if (D_8089E34C != -1) {
+                    this->actor.cutscene = D_8089E34C;
+                }
+            }
+
+            if (this->actor.cutscene != -1) {
+                func_801A2ED8();
+            }
+        }
+
+        func_8089ACEC(this, globalCtx);
+        func_8089AD70(this);
+        if (globalCtx->sceneNum == SCENE_MITURIN) {
+            globalCtx->envCtx.unk_C3 = 255;
+        }
+
+        this->colliderQuad.base.atFlags &= ~(AT_ON | AT_BOUNCED);
+        this->colliderJntSph.base.atFlags &= ~AT_ON;
+        if (this->actor.colChkInfo.damageEffect == 0x3) {
+            func_8089AC70(this);
+            if (this->actor.colChkInfo.health == 0) {
+                this->unk_290 = 3;
+                this->colliderJntSph.base.acFlags &= ~AC_ON;
+            }
+            func_8089C784(this);
+            return true;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 0x1) {
+            this->unk_290 = 40;
+            func_800BCB70(&this->actor, 0, 255, 0, 40);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_COMMON_FREEZE);
+            func_8089C784(this);
+            return true;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 0x5) {
+            this->unk_290 = 40;
+            func_800BCB70(&this->actor, 0, 255, 0, 40);
+            this->unk_2B4 = 0.55f;
+            this->unk_2B0 = 2.0f;
+            this->unk_28A = 31;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_COMMON_FREEZE);
+            func_8089C784(this);
+            return true;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 0x2) {
```

Comment:
```suggestion
        if (this->actor.colChkInfo.damageEffect == 2) {
```
to be consistent with how you're using bgcheck flags